content/movie-review-pt1.md

```diff
@@ -0,0 +1,464 @@
+
+To make deserializing instruction data simple, you can create a struct representing the data and use the `derive` attribute to apply the `BorshDeserialize` trait to the struct. This implements the methods defined in `BorshDeserialize`, including the the `try_from_slice` method that we'll be using to deserialize the instruction data.
+
+Remember, the struct itself needs to match the structure of the data in the byte array.
+
+```rust
+#[derive(BorshDeserialize)]
+struct NoteInstructionPayload {
+    id: u64,
+    title: String,
+    body: String
+}
+```
+
+Once this struct has been created, you can create an implementation for your instruction enum to handle the logic associated with deserializing instruction data. It's common to see this done inside a function called `unpack` that accepts the instruction data as an argument and returns the appropriate instance of the enum with the deserialized data.
+
+It's standard practice to structure your program to expect the first byte (or x number of bytes) to be an identifier for which instruction the program should run. This could be an integer or a string identifier. For this example, we'll use the first byte and map integers 0, 1, and 2 to instructions create, update, and delete, respectively.
+
+```rust
+impl NoteInstruction {
+    // Unpack inbound buffer to associated Instruction
+    // The expected format for input is a Borsh serialized vector
+    pub fn unpack(input: &[u8]) -> Result<Self, ProgramError> {
+        // Take the first byte as the variant to
+        // determine which instruction to execute
+        let (&variant, rest) = input.split_first().ok_or(ProgramError::InvalidInstructionData)?;
+        // Use the temporary payload struct to deserialize
+        let payload = NoteInstructionPayload::try_from_slice(rest).unwrap();
+        // Match the variant to determine which data struct is expected by
+        // the function and return the TestStruct or an error
+        Ok(match variant {
+            0 => Self::CreateNote {
+                title: payload.title,
+                body: payload.body,
+                id: payload.id
+            },
+            1 => Self::UpdateNote {
+                title: payload.title,
+                body: payload.body,
+                id: payload.id
+            },
+            2 => Self::DeleteNote {
+                id: payload.id
+            },
+            _ => return Err(ProgramError::InvalidInstructionData)
+        })
+    }
+}
+```
+
+This function starts by using the `split_first` function on the `input` parameter to return a tuple where the first element, `variant`, is the first byte from the byte array and the second element, `rest`, is the rest of the byte array. It then uses the `try_from_slice` method on `NoteInstructionPayload` to deserialize the rest of the byte array into an instane of `NoteInstructionPayload`. Finally, it uses a `match` statement on `variant` to create and return the appropriate enum instance.
```

Comment:
@jamesrp13 "intane" --> instance

Towards the end of the paragraph, "byte array into an instane of `NoteInstructionPayload`."